### ISSUE TITLE
System.ArgumentNullException thrown from ZipOutputStream when writing AES encrypted/Stored entries

### ISSUE THREAD

**Issue by Numpsy** [icsharpcode/SharpZipLib#322]
### Steps to reproduce
Try to use ZipOutputStream to create an AES encrypted entry whose compressionMethod is set to Stored.
e.g.
```
using (var memoryStream = new MemoryStream())
{
	using (var zipStream = new ZipOutputStream(memoryStream))
	{
		zipStream.IsStreamOwner = false;
		zipStream.Password = "password";  // optional. Null is the same as not setting. Required if using AES.

		ZipEntry zipEntry = new ZipEntry("stored");
		zipEntry.AESKeySize = 256;
		zipEntry.DateTime = DateTime.Now;
		zipEntry.CompressionMethod = CompressionMethod.Stored;

		zipStream.PutNextEntry(zipEntry);

		byte[] dummyData = Encoding.UTF8.GetBytes(DummyDataString);

		using (var dummyStream = new MemoryStream(dummyData))
		{
			dummyStream.CopyTo(zipStream);
		}

		zipStream.CloseEntry();
	}
}	
```

### Expected behavior
The entry should be added

### Actual behavior
The call to CloseEntry() throws an ArgumentNullException.

### Obtained from (only keep the relevant lines)
- Compiled from source, commit: 25ee644

**Comment by Numpsy** [comment]
Problem seems to be that the AESAuthCode property of the output stream isn't set up in this case, so the attempt to use it @ https://github.com/icsharpcode/SharpZipLib/blob/25ee6440f444aa326ee8f0b98e6223296f35fa6d/src/ICSharpCode.SharpZipLib/Zip/ZipOutputStream.cs#L493 throws.

See referenced PR for a possible fix.

**Comment by Numpsy** [comment]
Should be fixed by #323, so i'll close this now.
